### PR TITLE
Remove some unused methods.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Remove unused methods in `HistoryRecordManager` and other locations.
- Remove `OnConflictStrategy.ABORT` for the Room `Insert` annotation, as it is now the default value.

#### Before/After Screenshots/Screen Record

- Before:
- After:

#### Fixes the following issue(s)

- Fixes #

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/database/BasicDAO.java [hunk 1]
```diff
@@ -14,13 +13,10 @@
 @Dao
 public interface BasicDAO<Entity> {
     /* Inserts */
-    @Insert(onConflict = OnConflictStrategy.ABORT)
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/database/BasicDAO.java:17]: Why this change? Could you document such changes when you open a PR?

**Review by Stypox — CHANGES_REQUESTED** [review]
Other than the comment this looks good to me, I also wanted to do it

File: app/src/main/java/org/schabi/newpipe/database/BasicDAO.java (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/database/BasicDAO.java:17]: Sorry, my bad.

**Review by Stypox — APPROVED** [review]
> Remove OnConflictStrategy.ABORT for the Room Insert annotation, as it is now the default value.

Oh ok, according to [the source code blame](https://android.googlesource.com/platform/frameworks/support/+blame/android-room-release/room/common/src/main/java/androidx/room/Insert.java), `OnConflictStrategy.ABORT` has been the default at least since 2018.
